test/unit/test_borealis.py

```diff
@@ -0,0 +1,1031 @@
+
+
+        try:
+            writer = pydarn.BorealisSiteWrite(
+                "test_rawacf.rawacf.hdf5", self.rawacf_site_incorrect_fmt, 'rawacf')
+        except pydarn.borealis_exceptions.BorealisDataFormatTypeError as err:
+            self.assertEqual(
+                err.incorrect_types['scan_start_marker'], "<class 'numpy.bool_'>")
+            self.assertEqual(err.record_name, keys[0])
+
+    def test_writing_bfiq(self):
+        """
+        Tests write_bfiq method - writes a bfiq file
+
+        Expected behaviour
+        ------------------
+        bfiq file is produced
+        """
+        test_file = "./test_bfiq.bfiq.hdf5"
+        writer = pydarn.BorealisSiteWrite(test_file, 
+                                          self.bfiq_site_data, 'bfiq')
+        # only testing the file is created since it should only be created
+        # at the last step after all checks have passed
+        # Testing the integrity of the insides of the file will be part of
+        # integration testing since we need BorealisSiteRead for that.
+        self.assertTrue(os.path.isfile(test_file))
+        records = pydarn.read_borealis_file(test_file, 'bfiq', site=True, records=True)
+        self.check_dictionaries_are_same(records, self.bfiq_site_data)
+        os.remove(test_file)
+
+    def test_missing_field_bfiq(self):
+        """
+        Tests write_bfiq method - writes a bfiq structure file for the
+        given data
+
+        Expected behaviour
+        ------------------
+        Raises BorealisFieldMissingError - because the bfiq data is
+        missing field antenna_arrays_order
+        """
+        keys = sorted(list(self.bfiq_site_missing_field.keys()))
+        del self.bfiq_site_missing_field[keys[0]]['antenna_arrays_order']
+        
+        try:
+            writer = pydarn.BorealisSiteWrite(
+                "test_bfiq.bfiq.hdf5", self.bfiq_site_missing_field, 'bfiq')
+        except pydarn.borealis_exceptions.BorealisFieldMissingError as err:
+            self.assertEqual(err.fields, {'antenna_arrays_order'})
+            self.assertEqual(err.record_name, keys[0])
+
+    def test_extra_field_bfiq(self):
+        """
+        Tests write_bfiq method - writes a bfiq structure file for the
+        given data
+
+        Expected behaviour
+        ------------------
+        Raises BorealisExtraFieldError because the bfiq data
+        has an extra field dummy
+        """
+        keys = sorted(list(self.bfiq_site_extra_field.keys()))
+        self.bfiq_site_extra_field[keys[0]]['dummy'] = 'dummy'
+
+        try:
+            writer = pydarn.BorealisSiteWrite("test_bfiq.bfiq.hdf5", 
+                self.bfiq_site_extra_field, 'bfiq')
+        except pydarn.borealis_exceptions.BorealisExtraFieldError as err:
+            self.assertEqual(err.fields, {'dummy'})
+            self.assertEqual(err.record_name, keys[0])
+
+    def test_incorrect_data_format_bfiq(self):
+        """
```

Comment:
Try and make a descriptive, one line statement here, since many loggers will only print out the first line.